infra-examples/aws/main.tf

```diff
@@ -0,0 +1,91 @@
+locals {
+  kubernetes_version = "1.30"
+  cluster_name       = "${var.kubernetes_cluster_name}-${var.environment}"
+}
+
+data "aws_caller_identity" "current" {}
+
+data "aws_availability_zones" "available" {}
+
+module "main_vpc" {
+  source = "../../terraform/modules/aws/vpc"
+
+  environment        = var.environment
+  availability_zones = data.aws_availability_zones.available.names
+
+  single_nat_gateway     = true
+  one_nat_gateway_per_az = false
+
+  private_subnets = [
+    "10.0.1.0/24",
+    "10.0.2.0/24",
+    "10.0.3.0/24",
+  ]
+
+  public_subnets = [
+    "10.0.101.0/24",
+    "10.0.102.0/24",
+    "10.0.103.0/24",
+  ]
+
+  tags = {
+    "kubernetes.io/cluster/${local.cluster_name}" = "shared"
+  }
+
+  public_subnet_tags = {
+    "Tier"                                        = "Public"
+    "kubernetes.io/cluster/${local.cluster_name}" = "shared"
+    "kubernetes.io/role/elb"                      = "1"
+  }
+  private_subnet_tags = {
+    "Tier"                                        = "Private"
+    "kubernetes.io/cluster/${local.cluster_name}" = "shared"
+    "kubernetes.io/role/internal-elb"             = "1"
+  }
+}
+
+module "kubernetes_cluster" {
+  source     = "../../terraform/modules/aws/eks"
+
+  environment = var.environment
+  vpc_id      = module.main_vpc.vpc_id
+
+  cluster_name         = var.kubernetes_cluster_name
+  kubernetes_version   = local.kubernetes_version
+  registry_credentials = var.docker_registry_credentials
+
+  worker_node_ssh_key_name   = var.worker_node_ssh_key_name
+  worker_node_instance_types = ["m6i.large"]
+}
+
+module "bucket" {
+  source = "../../terraform/modules/aws/s3"
+
+  environment   = var.environment
+  bucket_prefix = "my-institute"
+}
```

Comment:
So if I understand correctly, this creates an S3 bucket called `my-institute`? Is there a use case for that? Why do we even have the `s3` module at all?

Reply:
Based on my best knowledge, Tutor is not creating an S3 bucket nor the tutor-contrib-s3. Therefore, someone have to create the bucket. We discussed having all modules available to create the infrastructure for running instances on a Kubernetes cluster. The S3 bucket is something needed for the instances during runtime, and we provide a module showcasing the CORS settings, bucket encryption, versioning, etc.

Reply:
I see. Hmm, this is not as separated as I thought. If we are using Terraform for per-instance stuff like S3 buckets and MongoDB databases, do you think we can/should at least have a separate Terraform state per instance rather than combining that all in with the overall cluster provisioning?

Reply:
I think this question/answer is highly related to [the one above](https://github.com/openedx/openedx-k8s-harmony/pull/143#discussion_r1905871474).